luxonis_ml/data/datasets/luxonis_dataset.py

```diff
@@ -531,36 +563,86 @@ def get_splits(self) -> Optional[Dict[str, List[str]]]:
 
     def make_splits(
         self,
-        ratios: Tuple[float, float, float] = (0.8, 0.1, 0.1),
-        definitions: Optional[Dict[str, List[PathType]]] = None,
+        ratios: Optional[Union[Dict[str, float], Tuple[float, float, float]]] = None,
+        definitions: Optional[Mapping[str, Sequence[PathType]]] = None,
+        replace_old_splits: bool = False,
     ) -> None:
-        new_splits = {"train": [], "val": [], "test": []}
-        splits_to_update = []
+        if ratios is not None and definitions is not None:
+            raise ValueError("Cannot provide both ratios and definitions")
+
+        if ratios is not None:
+            if isinstance(ratios, tuple):
+                if not len(ratios) == 3:
+                    raise ValueError(
+                        "Ratios must be a tuple of 3 floats for train, val, and test splits"
+                    )
+                ratios = {"train": ratios[0], "val": ratios[1], "test": ratios[2]}
+            sum_ = sum(ratios.values())
+            if not math.isclose(sum_, 1.0):
+                raise ValueError(f"Ratios must sum to 1.0, got {sum_:0.4f}")
+
+        if definitions is not None:
+            n_files = sum(map(len, definitions.values()))
+            if n_files > len(self):
+                raise ValueError(
+                    "Dataset size is smaller than the total number of files in the definitions. "
+                    f"Dataset size: {len(self)}, Definitions: {n_files}."
+                )
+
+        splits_to_update: List[str] = []
+        new_splits: Dict[str, List[str]] = {}
+        old_splits: Dict[str, List[str]] = defaultdict(list)
+
+        splits_path = self._get_file(
+            "metadata/splits.json",
+            self.local_path,
+            default=self.metadata_path / "splits.json",
+        )
+        assert splits_path is not None
+        if splits_path.exists():
+            with open(splits_path, "r") as file:
+                old_splits = defaultdict(list, json.load(file))
+
+        defined_uuids = set(uuid for uuids in old_splits.values() for uuid in uuids)
 
         if definitions is None:
+            ratios = ratios or {"train": 0.8, "val": 0.1, "test": 0.1}
             df = self._load_df_offline()
             assert df is not None
-            ids = df.select("uuid").unique().get_column("uuid").to_list()
+            ids = (
+                df.filter(~pl.col("uuid").is_in(defined_uuids))
+                .select("uuid")
+                .unique()
+                .get_column("uuid")
+                .to_list()
+            )
+            if not ids:
+                if not replace_old_splits:
+                    raise ValueError(
```

Comment:
Not sure what the default behavior should be when trying to make splits with no new data. Whether to raise an error or just log a warning and do nothing.

Reply:
I would say that this would be a "user error" and we should explicitly notify him of it as most of the time warnings aren't read.